src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,233 @@
-     * @param minLongitude
-     * @param maxLongitude
-     * @param minLatitude
-     * @param maxLatitude
-     * @return
-     */
-    public Collection<RoadAccident> getAccidentsByLocation(float minLongitude, float maxLongitude, float minLatitude, float maxLatitude){
-        return null;
-    }
-
-    /**
-     * find the weather conditions which caused max number of incidents
-     * @return
-     */
-    public List<String> getTopThreeWeatherCondition(){
-        return null;
-    }
-
-    /**
-     * count incidents by road surface conditions
-     * @return
-     */
-    public Map<String, Long> getCountByRoadSurfaceCondition(){
-        return null;
-    }
-
-    /**
-     * To match streaming operations result, return type is a java collection instead of multimap
-     * @return
-     */
-    public Map<String, List<String>> getAccidentIdsGroupedByAuthority(){
-        return null;
-    }
+	private final List<RoadAccident> roadAccidentList;
+
+	public DataProcessor(List<RoadAccident> roadAccidentList) {
+		this.roadAccidentList = roadAccidentList;
+	}
+
+	// First try to solve task using java 7 style for processing collections
+
+	/**
+	 * Return road accident with matching index
+	 * 
+	 * @param index
+	 * @return
+	 */
+	public RoadAccident getAccidentByIndex7(String index) {
+		RoadAccident roadAccident = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			for (RoadAccident accident : roadAccidentList) {
+				if (accident.getAccidentId() != null
+						&& accident.getAccidentId().equals(index)) {
+					roadAccident = accident;
+					break;
+				}
+			}
+		}
+		return roadAccident;
+	}
+
+	/**
+	 * filter list by longtitude and latitude values, including boundaries
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation7(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		Collection<RoadAccident> colls = null;
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			colls = new ArrayList<RoadAccident>();
+			for (RoadAccident accident : roadAccidentList) {
+				if (accident.getLongitude() <= maxLongitude
+						&& accident.getLongitude() >= minLongitude
+						&& accident.getLatitude() <= maxLatitude
+						&& accident.getLatitude() >= minLatitude) {
+					colls.add(accident);
+				}
+			}
+		}
+		return colls;
+	}
+
+	/**
+	 * count incidents by road surface conditions ex: wet -> 2 dry -> 5
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition7() {
+		//roadSurfaceConditions
+		Map<String, Long> resultMap = null;
+		
+		if (roadAccidentList != null && roadAccidentList.size() > 0) {
+			resultMap = new HashMap<String, Long>(); 
+			Multimap<String, RoadAccident> multiMap = ArrayListMultimap.create();
+			for (RoadAccident accident : roadAccidentList) {
+				multiMap.put(accident.getRoadSurfaceConditions(),
```

Comment:
Interesting use of multimap. I was thinking more about using counter right away. But this one is interesting :+1: